app/src/main/java/fr/neamar/kiss/dataprovider/AppProvider.java

```diff
@@ -125,31 +128,119 @@ public void requestResults(String query, Searcher searcher) {
             return;
         }
 
-        FuzzyScore fuzzyScore = FuzzyFactory.createFuzzyScore(this, queryNormalized.codePoints);
+        boolean flexibleFuzzy = prefs.getBoolean("enable-fuzzy-search", false);
+
+        if (flexibleFuzzy) {
+            // New flexible fuzzy search logic for partial, misspelled queries
+            for (AppPojo pojo : getPojos()) {
+                // exclude apps from results
+                if (pojo.isExcluded() && !prefs.getBoolean("enable-excluded-apps", false)) {
+                    continue;
+                }
+                // exclude favorites from results
+                if (excludedFavoriteIds.contains(pojo.getFavoriteId())) {
+                    continue;
+                }
+
+                // Match against app name
+                int distance = substringLevenshteinDistance(queryNormalized.codePoints, pojo.normalizedName.codePoints);
```

Comment:
I wonder how this behaves in terms of performance with 5k contacts. I remember Levenshtein isn't computationally free, so I'll be curious to test this with live data :)

Reply:
i think jaro/jaro-winkler is faster